pkg/apis/kops/validation/instancegroup.go

```diff
@@ -105,3 +114,29 @@ func CrossValidateInstanceGroup(g *kops.InstanceGroup, cluster *kops.Cluster, st
 
 	return nil
 }
+
+func validateExtraUserData(userData *kops.ExtraUserDataSpec) error {
+	if userData.Name == "" {
+		return field.Required(field.NewPath("ExtraUserData Name"), "extraUserData Name field must be set")
+	}
+
+	if userData.Content == "" {
+		return field.Required(field.NewPath("ExtraUserData Content"), "extraUserData Content field must be set")
+	}
+
+	switch userData.Type {
+	case "text/x-include-once-url":
+	case "text/x-include-url":
+	case "text/cloud-config-archive":
+	case "text/upstart-job":
+	case "text/cloud-config":
+	case "text/part-handler":
+	case "text/x-shellscript":
+	case "text/cloud-boothook":
```

Comment:
The only one I need personally is "text/cloud-config". Just though I'd add all the valid types to have inclusive support since we're already putting support in. However I have nothibg against removing x-shellscript.

Reply:
I think we probably should, for safety, but I also want to get this PR in, so maybe we look at that in a separate PR :-)

Reply:
(We can always add things to the set of permitted types, much harder to remove things)